### PR TITLE
.Net: Add Data ingestion sample for VectorStore

### PR DESCRIPTION
**Pull request by westey-m**
### Motivation and Context

Adding a sample that uses the new VectorStore implementations.

Alternative suggestions for the best sample category for these are welcome.  For now they are in the Memory folder.
I have not updated the samples README.md yet, and will update, once we have a plan around where these should go.

#7349

### Contribution Checklist

- [x] The code builds clean without any errors or warnings
- [x] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [x] All unit tests pass, and I have added new tests where possible
- [x] I didn't break anyone :smile:

### CODE REVIEW

File: dotnet/samples/Concepts/Memory/VectorStoreFixtures/VectorStore_QdrantContainer_Fixture.cs [hunk 1]
```diff
@@ -0,0 +1,31 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using Docker.DotNet;
+
+namespace Memory.VectorStoreFixtures;
+
+/// <summary>
+/// Fixture to use for creating a Qdrant container before tests and delete it after tests.
+/// </summary>
+public class VectorStore_QdrantContainer_Fixture : IAsyncLifetime
```
Comment (dmytrostruk) [inline dotnet/samples/Concepts/Memory/VectorStoreFixtures/VectorStore_QdrantContainer_Fixture.cs:10]: Is there any reason to use underscores in class names here?

File: dotnet/samples/Concepts/Memory/VectorStore_DataIngestion.cs [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+public class VectorStore_DataIngestion(ITestOutputHelper output) : BaseTest(output), IClassFixture<VectorStore_RedisContainer_Fixture>, IClassFixture<VectorStore_QdrantContainer_Fixture>
+{
+    /// <summary>
+    /// Main entry point for example.
+    /// </summary>
+    /// <param name="databaseType">The type of database to run the example for.</param>
+    [Theory]
+    [InlineData("Redis")]
+    [InlineData("Qdrant")]
+    [InlineData("Volatile")]
```
Comment (dmytrostruk) [inline dotnet/samples/Concepts/Memory/VectorStore_DataIngestion.cs:36]: nit: I would probably keep these examples in separate methods. For me it's easier to run it when it's separated, but it's just personal preference, so feel free to skip.

**Review by dmytrostruk — APPROVED** [review]

File: dotnet/samples/Concepts/Memory/VectorStore_DataIngestion.cs [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+/// Since Redis and Volatile supports string keys and Qdrant supports ulong or Guid keys, this example also shows how you can have common code
+/// that works with both types of keys by using a generic key generator function.
+///
+/// The example shows the following steps:
+/// 1. Register a vector store and embedding generator with the DI container.
+/// 2. Register a class (DataIngestor) with the DI container that uses the vector store and embedding generator to ingest data.
+/// 3. Ingest some data into the vector store.
+/// 4. Read the data back from the vector store.
+///
+/// To run this sample, you need a local instance of Docker running, since the associated fixtures will try and start Redis and Qdrant containers in the local docker instance.
```
[the remark this answers is not in the document]
Comment (westey-m) [reply dotnet/samples/Concepts/Memory/VectorStore_DataIngestion.cs:25]: OK, thanks, makes sense.

File: dotnet/samples/Concepts/Memory/VectorStore_DataIngestion.cs [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+        // Register the chosen vector store with the DI container.
+        if (databaseType == "Redis")
+        {
+            kernelBuilder.AddRedisVectorStore("localhost:6379");
+        }
+        else if (databaseType == "Qdrant")
+        {
+            kernelBuilder.AddQdrantVectorStore("localhost");
+        }
+        else if (databaseType == "Volatile")
```
[the remark this answers is not in the document]
Comment (westey-m) [reply dotnet/samples/Concepts/Memory/VectorStore_DataIngestion.cs:58]: Happy to chat more about it.  I considered InMemory, but I was concerned about any confusion considering the usage of the term memory in the same context already.

File: dotnet/samples/Concepts/Memory/VectorStoreFixtures/VectorStore_QdrantContainer_Fixture.cs (hunk 1 above)
Comment (westey-m) [reply dotnet/samples/Concepts/Memory/VectorStoreFixtures/VectorStore_QdrantContainer_Fixture.cs:10]: No, I can change it.  The samples all have underscores, making it easier to read their names, but I suppose these are not samples, and they are in their own folder, so I'll just change them.

File: dotnet/samples/Concepts/Memory/VectorStore_DataIngestion.cs (hunk 2 above)
Comment (westey-m) [reply dotnet/samples/Concepts/Memory/VectorStore_DataIngestion.cs:36]: I was actually quite keen to show how the code is all the same except for the DI registration and the function to generate the unique key.  If I split them out, I don't think it'll be as obvious, so I'd rather keep it this way.

**Review by markwallace-microsoft — APPROVED** [review]

**Pull request merged**